libs/libcommon/src/libcommon/orchestrator.py

```diff
@@ -647,253 +667,196 @@ class DatasetRemovalPlan(Plan):
-        plan.run()
-
-    def set_revision(
-        self, revision: str, priority: Priority, error_codes_to_retry: list[str], cache_max_days: int
-    ) -> None:
-        """
-        Set the current revision of the dataset.
+    If the revision is already set to the same value, this is a no-op. Else: one job is created for every first
+        step.
 
-        If the revision is already set to the same value, this is a no-op. Else: one job is created for every first
-          step.
+    Args:
+        dataset (str): The name of the dataset.
+        revision (str): The new revision of the dataset.
+        priority (Priority): The priority of the jobs to create.
+        cache_max_days (int): The maximum number of days for which the cache is considered valid.
+        error_codes_to_retry (list[str], optional): The error codes for which the jobs should be retried.
+        processing_graph (ProcessingGraph, optional): The processing graph.
 
-        Args:
-            revision (str): The new revision of the dataset.
-            priority (Priority): The priority of the jobs to create.
-            error_codes_to_retry (list[str]): The error codes for which the jobs should be retried.
-            cache_max_days (int): The maximum number of days for which the cache is considered valid.
+    Returns:
+        None
 
-        Returns:
-            None
-
-        Raises:
-            ValueError: If the first processing steps are not dataset steps, or if the processing graph has no first
-              step.
-        """
-        self._raise_and_remove_if_blocked()
-        first_processing_steps = self.processing_graph.get_first_processing_steps()
-        if len(first_processing_steps) < 1:
-            raise ValueError("Processing graph has no first step")
-        if any(first_processing_step.input_type != "dataset" for first_processing_step in first_processing_steps):
-            raise ValueError("One of the first processing steps is not a dataset step")
+    Raises:
+        ValueError: If the first processing steps are not dataset steps, or if the processing graph has no first
+            step.
+    """
+    first_processing_steps = processing_graph.get_first_processing_steps()
+    if len(first_processing_steps) < 1:
+        raise ValueError("Processing graph has no first step")
+    if any(first_processing_step.input_type != "dataset" for first_processing_step in first_processing_steps):
+        raise ValueError("One of the first processing steps is not a dataset step")
+    with StepProfiler(
+        method="set_revision",
+        step="all",
+        context=f"dataset={dataset}",
+    ):
+        logging.info(f"Analyzing {dataset}")
         with StepProfiler(
-            method="DatasetOrchestrator.set_revision",
-            step="all",
-            context=f"dataset={self.dataset}",
+            method="set_revision",
+            step="plan",
+            context=f"dataset={dataset}",
```

Comment:
Should we stop passing "context" ? It makes Grafana unusable since there are so many datasets

Reply:
Makes sense, lets clean that in another PR. We can pass what we want, really it's up to you if it helps. Feel free to open an issue.